lib/engine/game/g_18_carolinas/game.rb

```diff
@@ -410,7 +510,46 @@ def upgrades_to?(from, to, special = false, selected_company: nil)
           # handle C tiles specially
           return false if from.label.to_s == 'C' && to.color == :yellow && from.cities.size != to.cities.size
 
-          super
+          # correct color progression?
+          return false unless Engine::Tile::COLORS.index(to.color) == (Engine::Tile::COLORS.index(from.color) + 1)
```

Comment:
again, don't copy and paste huge chunks of code like this

Reply:
OK, I'll make this check overriddable.